File: NorthAx/Models/TrainingDomain.swift
import SwiftUI

enum TrainingDomain: String, Codable, CaseIterable, Identifiable {
    case cycling   = "Cycling"
    case running   = "Running"
    case strength  = "Strength"
    case swimming  = "Swimming"
    case triathlon = "Triathlon"
    case mobility  = "Mobility"
    case recovery  = "Recovery"

    var id: String { rawValue }

    var icon: String {
        switch self {
        case .cycling:   return "bicycle"
        case .running:   return "figure.run"
        case .strength:  return "dumbbell"
        case .swimming:  return "figure.pool.swim"
        case .triathlon: return "trophy"
        case .mobility:  return "figure.flexibility"
        case .recovery:  return "heart.circle"
        }
    }

    var color: Color {
        switch self {
        case .cycling:   return .axAccent
        case .running:   return .axGreen
        case .strength:  return .axRed
        case .swimming:  return .axBlue
        case .triathlon: return .axPurple
        case .mobility:  return Color(red: 0.9, green: 0.7, blue: 0.2)
        case .recovery:  return Color(red: 0.3, green: 0.8, blue: 0.7)
        }
    }
}

File: android/app/src/main/java/app/northax/domain/model/TrainingDomain.kt
package app.northax.domain.model

import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.DirectionsBike
import androidx.compose.material.icons.automirrored.filled.DirectionsRun
import androidx.compose.material.icons.filled.EmojiEvents
import androidx.compose.material.icons.filled.Favorite
import androidx.compose.material.icons.filled.FitnessCenter
import androidx.compose.material.icons.filled.Pool
import androidx.compose.material.icons.filled.SelfImprovement
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import app.northax.ui.theme.Ax

/** Raw values match the backend wire strings ("Cycling", "Running", …). */
enum class TrainingDomain(val raw: String) {
    Cycling("Cycling"),
    Running("Running"),
    Strength("Strength"),
    Swimming("Swimming"),
    Triathlon("Triathlon"),
    Mobility("Mobility"),
    Recovery("Recovery");

    val icon: ImageVector
        get() = when (this) {
            Cycling -> Icons.AutoMirrored.Filled.DirectionsBike
            Running -> Icons.AutoMirrored.Filled.DirectionsRun
            Strength -> Icons.Filled.FitnessCenter
            Swimming -> Icons.Filled.Pool
            Triathlon -> Icons.Filled.EmojiEvents
            Mobility -> Icons.Filled.SelfImprovement
            Recovery -> Icons.Filled.Favorite
        }

    val color: Color
        get() = when (this) {
            Cycling -> Ax.Cycling
            Running -> Ax.Green
            Strength -> Ax.StrengthSport
            Swimming -> Ax.Blue
            Triathlon -> Ax.Purple
            Mobility -> Ax.Amber
            Recovery -> Ax.Recovery
        }

    companion object {
        fun fromRaw(raw: String): TrainingDomain? = entries.firstOrNull { it.raw == raw }
    }
}
